lib/common/xml.c

```diff
@@ -3174,21 +3171,22 @@ crm_xml_dump(xmlNode * data, int options, char **buffer, int *offset, int *max,
          */
         xmlBufferSetAllocationScheme(xml_buffer, XML_BUFFER_ALLOC_DOUBLEIT);
 
-        *max = xmlNodeDump(xml_buffer, doc, data, 0, (options & xml_log_option_formatted));
-        if (*max > 0) {
-            *buffer = strdup((char *)xml_buffer->content);
+        ret = xmlNodeDump(xml_buffer, doc, data, 0, (options & xml_log_option_formatted));
```

Comment:
It's actually next to impossible unless one wants to fiddle with
struct fields and make some assumptions about the invariants, since
`xmlBufCreate` is a private API.  The respective code comment advice
that you reproduced here is likely wrong (or comes from when it was
undecided or something):

https://mail.gnome.org/archives/xml/2019-July/msg00009.html

Reply:
See the first commit.